doc/contributor/howto-guide-setup-development-workstation.md

```diff
@@ -123,6 +123,46 @@ alternatives.
 ./ci/install-cloud-sdk.sh
 ```
 
+If you use an editor that uses `compile_commands.json` for autocomplete, one
+can generate it with CMake using
+[CMAKE_EXPORT_COMPILE_COMMANDS](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html).
+If you prefer using Bazel, the following steps will generate one for you.
+
+First, install Bazelisk:
+
+```console
+go get github.com/bazelbuild/bazelisk
+# Alternatively, if you have Go 1.17 or later
+go install github.com/bazelbuild/bazelisk@latest
+```
+
+Add Go binaries to your PATH:
+
+```console
+export PATH=$PATH:$(go env GOPATH)/bin
```

Comment:
Thanks for the PR @remyabel. I like this. I'm just wondering if we want to just commit this script to our repo so it's even easier for folks to run. The instructions could just be: "To generate a `compile_commands.json` file run `ci/compile-commands.sh`. Re-run that command whenever files are added or deleted from the repo." The body of the script would be pretty much what you have now, except it could also install bazelisk iff needed.

Note: we could also include the link to https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html as well, though I believe using bazel to generate the compile commands file would be fine even if you use cmake to build the code, but maybe someone (@coryan ?) could confirm that.

Reply:
I guess. It would add a maintenance burden and the script is more of an example than authoritative steps. I copy/pasted the script you wrote (see the original issue). Would have to evaluate what the easiest option for everyone is. This happens to be the easiest option for me now. Overtime as the project developed, the instructions for generating `compile_commands.json` were removed, then the ability to add extra cmake commands to docker were removed (and not added back yet, I think I saw an issue on that) and as mentioned in another comment having a separate build just for it is kind of unwieldly as it requires compiling the entire project and that takes forever. I know ninja has something like `compdb`, but that requires fiddling. And when running it inside Docker, the paths are messed up. The current approach seems quickest and fastest so far and least likely to become obsolete as the code base changes overtime IMO.